### DISCUSSION TITLE
Await until a tween is finished before starting the dialogue

### DISCUSSION THREAD

**Discussion by juliohq** [dialogic-godot/dialogic#2323] [Q&A] [answered]
I'm trying to animate a dialogue layer with a Tween (the width of a `PanelContainer` so that I can add an open/close effect when the speaker changes) but the timeline starts at the same time as the Tween starts. Is there any way to make Dialogic await until my animation is finished before starting? Is it related to [this PR](https://github.com/dialogic-godot/dialogic/pull/2151) or could it be helpful somehow when merged?

I'm using Godot v4.2.2-stable and the latest commit of Dialogic.

**Comment by CakeVR** [comment]
That's my PR, it adds a new Timeline Event that allows to construct a Tween. Feel free to give it a try.

You can create a method on an Autoload and await inside it. Once you use the Call Timeline Event, Dialogic will pause the timeline until the method returns.

**Reply by juliohq** [reply]
That looks really interesting and Dialogic indeed would benefit from having such events. I'm looking forward to it!

**Comment by Jowan-Spooner** [comment, marked as answer]
You can insert animations for showing and hiding the textbox (see the default textbox layers animation script). However it might be good to add an additional dialog_begin and dialog_end animation.

For now you probably can just set `Dialogic.paused` to true in _ready of one of your layers and then animate before setting Dialogic.paused to false again.

**Reply by juliohq** [reply]
That works fine, though I couldn't find a way to prevent my textbox layer to hide and play my hide animation instead. What would be the recommended way to do that?

**Reply by Jowan-Spooner** [reply]
Do you mean at the end of the timeline?

**Reply by juliohq** [reply]
Yes, after the timeline ends Dialogic should emit the `animation_textbox_hide` signal, which for some weird reason doesn't actually calls my `_textbox_hide` function connected to that signal (maybe it isn't connected in time?), which then plays the hide animation.

**Reply by Jowan-Spooner** [reply]
Hey @juliohq sorry again for the delay 🙈. I think it's most likely that this is just the timeline being deleted to quickly. To fix this you can just add a `[wait time="1"]` one second wait event, which will hide the textbox and delay the actual deletion of the style a bit.

**Reply by juliohq** [reply]
Is there any other way to make Dialogic wait without having to add wait events to each timeline? The Visual Novel style, for example, doesn't need wait events for it to work properly.

**Reply by Jowan-Spooner** [reply]
That's not true I think, the visual novel layout is also just deleted on timeline end and thus has no textbox hide animation if there is no time left. 
I cannot think of another way to do this right now (besides modifying the plugin code, e.g. inserting something in the end_timeline() method). It's something I will think about, because I think it's a common need, but it's not there yet.
